### ISSUE TITLE
Post a Code of Conduct and prepare to enforce it

### ISSUE THREAD

**Issue by flyingzumwalt** [ipfs/conf#46]
The ipfs CoC is not configured for in-person events. We need to publish a variant that addresses physical space, physical safety, reporting incidents, duty officers, etc.

I have seen this one used well https://github.com/code4lib/code-of-conduct/blob/master/code_of_conduct.md

**Comment by meiqimichelle** [comment]
Related: need something to indicate organizers on lanyards, and something to indicate duty officers, and a burner phone for them to use.

**Comment by flyingzumwalt** [comment]
I realized that my link didn't point to all the relevant info. Beyond posting a code of conduct, we should do a lot of the procedures listed in https://github.com/code4lib/code-of-conduct/blob/master/procedures.md

## Before the conference

* Assemble conference sub-comittee to handle implementation of procedures.
* Ensure Local Planning commitee includes a quiet, private space suitable for taking reports in venue requirements.
* Select duty officers, and familiarize them with this document and the CodeOfConduct4Lib.
* Identify an organize any required traning for duty officers.
* Decide how conferencegoers will be able to recognize duty officers, and publicize this.
* Schedule duty officer shifts.
* Ensure that leaders of official off-site events (tours / newcomer dinners, etc) are familar with CodeOfConduct4Lib. 
* Decide which conference organizers will be responsible for incident response, and familiarize them with this document and the CodeOfConduct4Lib.
* Make sure all of the people in this section have each others' contact information.
* Consult last year's organizers for any knowledge they need to transfer relevant to enforcement (e.g. prior incident response; iterations to procedure; or people with ongoing sanctions).
* Investigate local laws that may affect procedures. 

Your duty officers and conference organizers should be a small group of kind, articulate, and trustworthy people who reflect the diversity of the conference.

**Comment by diasdavid** [comment]
Some successful Code of Conduct examples:

- http://confcodeofconduct.com/
- http://2014.lxjs.org/code-of-conduct/
- https://2018.jsconf.eu/code-of-conduct/

**Comment by meiqimichelle** [comment]
This is next on my todo list. @flyingzumwalt where did we land on getting a burner phone? I can’t remember.

**Comment by meiqimichelle** [comment]
As discussed on our Friday (yesterday) check-in, I'm going to edit our existing IFPS community Code of Conduct (https://github.com/ipfs/community/blob/master/code-of-conduct.md) to include guidance for events, and submit a PR for review.

**Comment by meiqimichelle** [comment]
Done. @flyingzumwalt ready for your review here: https://github.com/ipfs/community/pull/325 .

We may also want to add another document for procedures, like Code4lib (https://github.com/code4lib/code-of-conduct/blob/master/procedures.md). What say you @flyingzumwalt ? It is nice to have these easily-accessible for volunteers.

**Comment by flyingzumwalt** [comment]
I tweaked what @meiqimichelle added and merged it. Thank you @meiqimichelle!

https://github.com/ipfs/community/blob/master/code-of-conduct-procedures-for-events.md

I merged it without waiting for approval of the PR because we need to have _something_ in place before tomorrow morning. I would have preferred to follow proper community process for changing something like the CoC, so I would be really happy if we follow up after this week to properly update the CoC, refine the parts that I added, and possibly adding more guidance for people who want to report an incident, like the guidance in the conflict resolution section of the Code4Lib CoC — I left it out this time for the sake of simplicity -- that stuff was included in @meiqimichelle's PR.

@meiqimichelle also flagged that the PyCon Code of Conduct has good policies that might be necessary to handle the scale of events like IPFS Conf. We should keep looking into that in advance of November.
